ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/SignatureFixUtil.fs

```diff
@@ -0,0 +1,110 @@
+open JetBrains.ReSharper.Psi.Tree
+
+let getRecordRepresentation (typeDecl: IFSharpTypeDeclaration) =
+    match typeDecl.TypeRepresentation with
+    | :? IRecordRepresentation as rr -> Some rr
+    | _ -> None
+
+let getSignatureRecordRepr (implementationRecordRepr: IRecordRepresentation) =
+    let decl = implementationRecordRepr.TypeDeclaration.DeclaredElement
+    let declarations = if isNull decl then Seq.empty else decl.GetDeclarations()
+
+    declarations
+    |> Seq.tryPick (fun d ->
+        match d with
+        | :? IFSharpTypeDeclaration as signatureTypeDecl when
+            signatureTypeDecl.GetSourceFile().IsFSharpSignatureFile ->
+            getRecordRepresentation signatureTypeDecl
+        | _ -> None)
+
+let getFieldType (rfd:IRecordFieldDeclaration) =
+    if isNull rfd then None else
+    let symbolUse = rfd.GetFcsSymbolUse()
+    match symbolUse.Symbol with
+    | :? FSharpField as ff -> Some (ff.FieldType, symbolUse.DisplayContext)
+    | _ -> None
+
+let mkRecordFieldDeclaration isMutable (implFieldDecl: IRecordFieldDeclaration) (implementationFieldType: FSharpType, displayContext) =
+    let factory = implFieldDecl.CreateElementFactory()
+    let typeUsage = factory.CreateTypeUsage(implementationFieldType.Format displayContext)
+    factory.CreateRecordFieldDeclaration(isMutable, implFieldDecl.DeclaredName, typeUsage)
+
+let updateSignatureFieldDecl (implFieldDecl: IRecordFieldDeclaration) (signatureFieldDecl: IRecordFieldDeclaration) =
+    let fieldTypeAreEqual =
+        let signatureFieldType = getFieldType signatureFieldDecl
+        let implementationFieldType =  getFieldType implFieldDecl
+        match implementationFieldType, signatureFieldType with
+        | None, None
+        | Some _, None
+        | None, Some _ -> false
+        | Some (i, _), Some (s, _) -> i = s
+
+    let isImplMutable = isNotNull implFieldDecl.MutableKeyword
+    let mutableAreEqual = isImplMutable = isNotNull signatureFieldDecl.MutableKeyword
+    
+    if implFieldDecl.SourceName = signatureFieldDecl.SourceName && fieldTypeAreEqual && mutableAreEqual then
+        // fields are identical
+        ()
+    elif implFieldDecl.SourceName <> signatureFieldDecl.SourceName then
+        // field names are different, update signature field name
+        signatureFieldDecl.SetName(implFieldDecl.NameIdentifier.Name, ChangeNameKind.SourceName)
+    else
+        let implementationFieldType = getFieldType implFieldDecl
+        match implementationFieldType with
+        | None -> ()
+        | Some tu ->
+
+        if not mutableAreEqual then
+            let updatedSignatureField = mkRecordFieldDeclaration isImplMutable implFieldDecl tu
+            ModificationUtil.ReplaceChild(signatureFieldDecl, updatedSignatureField)
+            |> ignore
+        else
+            let factory = signatureFieldDecl.CreateElementFactory()
+            let t,d = tu
+            let updatedTypeUsage = factory.CreateTypeUsageForSignature(t.Format d)
```

Comment:
Could you add tests covering different `open`s between impl and sig files? E.g. when a sig doesn't have some needed `open` or when it has `open` that would allow short type usage then in the impl file.